hub/integrations/tensorflow.py

```diff
@@ -0,0 +1,32 @@
+from hub.util.exceptions import ModuleNotInstalledException
+
+
+def dataset_to_tensorflow(dataset):
+    """Converts the dataset into a tensorflow compatible format"""
+    global tf
+    try:
+        import tensorflow as tf
+    except ModuleNotFoundError:
+        raise ModuleNotInstalledException(
+            "'tensorflow' should be installed to convert the Dataset into tensorflow format"
+        )
+
+    def tf_gen():
+        for index in range(len(dataset)):
+            sample = {}
+            for key in dataset.tensors:
+                value = dataset[key][index].numpy()
+                sample[key] = value
+            yield sample
+
+    def generate_signature():
+        signature = {}
+        for key in dataset.tensors:
+            dtype = dataset[key].meta["dtype"]
+            shape = dataset[key].shape
+            dynamic_shape = [None for dim in shape.lower]
+            signature[key] = tf.TensorSpec(shape=dynamic_shape, dtype=dtype)
+        return signature
+
+    signature = generate_signature()
+    return tf.data.Dataset.from_generator(tf_gen, output_signature=signature)
```

Comment:
Is there another layer, or is this the last step before training? If this is it, we should configure some prefetching defaults and such for free performance wins

Reply:
Do you mean this https://www.tensorflow.org/api_docs/python/tf/data/Dataset#prefetch?
Shouldn't we leave this up to the user? We're anyway telling them that this is a tf.data.Dataset object, so anything applicable to that can be used.